src/Sylius/Bundle/ApiBundle/Exception/ProductCannotBeRemoved.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Exception;
+
+/** @experimental */
+final class ProductCannotBeRemoved extends \RuntimeException
+{
+    public function __construct()
+    {
+        parent::__construct('Cannot delete a product that is in use by a promotion rule.');
```

Comment:
Gonna be a bit nitpicky, but all the other `*CannotBeRemoved` exceptions are phrased as `Cannot delete, the X is in use.`, why is this one so specific? Also, judging solely by the name of it I'd assume it can be used in other places where a product cannot be removed, but then the message would be just confusing.